msbuild/Xamarin.iOS.Tasks.Core/Xamarin.iOS.Common.targets

```diff
@@ -524,34 +524,61 @@ Copyright (C) 2013-2016 Xamarin. All rights reserved.
 		</Touch>
 	</Target>
 
-	<Target Name="_GenerateFrameworkDebugSymbols" Condition="'$(ComputedPlatform)' == 'iPhone' And '$(IsWatchApp)' == 'false'" DependsOnTargets="_CompileToNative;_ParseBundlerArguments;_CollectFrameworks"
-		Inputs="%(_Frameworks.Identity)"
-		Outputs="$(AppBundleDir)\..\%(_Frameworks.Filename).framework.dSYM\Contents\Info.plist"
+	<Target Name="_CopyUserFrameworkDebugSymbols" Condition="'$(ComputedPlatform)' == 'iPhone' And '$(IsWatchApp)' == 'false'" DependsOnTargets="_CompileToNative;_ParseBundlerArguments;"
+		Inputs="%(_FrameworkNativeReference.Name)"
+		Outputs="$(AppBundleDir)\..\%(_FrameworkNativeReference.Filename).framework.dSYM\Contents\Info.plist"
 		>
 
-		<RemoveDir SessionId="$(BuildSessionId)" Condition="'$(IsMacEnabled)' == 'true'" Directories="$(AppBundleDir)\..\%(_Frameworks.Filename).framework.dSYM" />
+		<PropertyGroup>
+			<_Source>%(_FrameworkNativeReference.Name)\..\dSYMs\%(_FrameworkNativeReference.Filename).framework.dSYM\</_Source>
+			<_Destination>$(AppBundleDir)\..\%(_FrameworkNativeReference.Filename).framework.dSYM</_Destination>
+		</PropertyGroup>
 
-		<!-- run dsymutil on embedded frameworks -->
-		<DSymUtil
-			SessionId="$(BuildSessionId)"
-			Condition="'$(IsMacEnabled)' == 'true' And '$(_NoDSymUtil)' == 'false'"
-			AppBundleDir="$(AppBundleDir)"
-			Architectures=""
-			DSymDir="$(AppBundleDir)\..\%(_Frameworks.Filename).framework.dSYM"
-			Executable="%(_Frameworks.Identity)"
-			ToolExe="$(DSymUtilExe)"
-			ToolPath="$(DSymUtilPath)">
-		</DSymUtil>
+		<RemoveDir SessionId="$(BuildSessionId)" Condition="'$(IsMacEnabled)' == 'true'" Directories="$(_Destination)" />
```

Comment:
I'm not sure this will work correctly if there are multiple user frameworks with .dSYMs, I think the value of _Source and _Destination will be for the last user framework processed.

I think something like this would work:

```xml
<ItemGroup>
    <_FrameworkNativeReference>
        <Source>%(_FrameworkNativeReference.Name)\..\dSYMs\%(_FrameworkNativeReference.Filename).framework.dSYM\</Source>
        <Destination>$(AppBundleDir)\..\%(_FrameworkNativeReference.Filename).framework.dSYM</Destination>
    </_FrameworkNativeReference>
</ItemGroup>
```

Reply:
Thanks, I was not sure about that - but did plan to add tests with more than one
They might come in a separate PR since there's more things to fix too.

Reply:
Need to double check but it what's already there seems to work
```
_CopyUserFrameworkDebugSymbols:
  Removing directory "bin/iPhone/Release/xcf_ios.app/../Universal.framework.dSYM".
  /usr/bin/ditto -rsrc ../../Universal.xcframework/ios-arm64/Universal.framework/../dSYMs/Universal.framework.dSYM/ bin/iPhone/Release/xcf_ios.app/../Universal.framework.dSYM 
_CopyUserFrameworkDebugSymbols:
  Removing directory "bin/iPhone/Release/xcf_ios.app/../wavpack.framework.dSYM".
  /usr/bin/ditto -rsrc ../../external/wavpack.xcframework/ios-arm64/wavpack.framework/../dSYMs/wavpack.framework.dSYM/ bin/iPhone/Release/xcf_ios.app/../wavpack.framework.dSYM 
_CopyUserFrameworkDebugSymbols:
  Removing directory "bin/iPhone/Release/xcf_ios.app/../vorbis.framework.dSYM".
  /usr/bin/ditto -rsrc ../../external/vorbis.xcframework/ios-arm64/vorbis.framework/../dSYMs/vorbis.framework.dSYM/ bin/iPhone/Release/xcf_ios.app/../vorbis.framework.dSYM 
_CopyUserFrameworkDebugSymbols:
  Removing directory "bin/iPhone/Release/xcf_ios.app/../tta++.framework.dSYM".
  /usr/bin/ditto -rsrc ../../external/tta++.xcframework/ios-arm64/tta++.framework/../dSYMs/tta++.framework.dSYM/ bin/iPhone/Release/xcf_ios.app/../tta++.framework.dSYM 
_CopyUserFrameworkDebugSymbols:
  Removing directory "bin/iPhone/Release/xcf_ios.app/../taglib.framework.dSYM".
  /usr/bin/ditto -rsrc ../../external/taglib.xcframework/ios-arm64/taglib.framework/../dSYMs/taglib.framework.dSYM/ bin/iPhone/Release/xcf_ios.app/../taglib.framework.dSYM 
_CopyUserFrameworkDebugSymbols:
  Removing directory "bin/iPhone/Release/xcf_ios.app/../speex.framework.dSYM".
  /usr/bin/ditto -rsrc ../../external/speex.xcframework/ios-arm64/speex.framework/../dSYMs/speex.framework.dSYM/ bin/iPhone/Release/xcf_ios.app/../speex.framework.dSYM 
_CopyUserFrameworkDebugSymbols:
  Removing directory "bin/iPhone/Release/xcf_ios.app/../sndfile.framework.dSYM".
  /usr/bin/ditto -rsrc ../../external/sndfile.xcframework/ios-arm64/sndfile.framework/../dSYMs/sndfile.framework.dSYM/ bin/iPhone/Release/xcf_ios.app/../sndfile.framework.dSYM 
_CopyUserFrameworkDebugSymbols:
  Removing directory "bin/iPhone/Release/xcf_ios.app/../opus.framework.dSYM".
  /usr/bin/ditto -rsrc ../../external/opus.xcframework/ios-arm64/opus.framework/../dSYMs/opus.framework.dSYM/ bin/iPhone/Release/xcf_ios.app/../opus.framework.dSYM 
_CopyUserFrameworkDebugSymbols:
  Removing directory "bin/iPhone/Release/xcf_ios.app/../ogg.framework.dSYM".
  /usr/bin/ditto -rsrc ../../external/ogg.xcframework/ios-arm64/ogg.framework/../dSYMs/ogg.framework.dSYM/ bin/iPhone/Release/xcf_ios.app/../ogg.framework.dSYM 
_CopyUserFrameworkDebugSymbols:
  Removing directory "bin/iPhone/Release/xcf_ios.app/../mpg123.framework.dSYM".
  /usr/bin/ditto -rsrc ../../external/mpg123.xcframework/ios-arm64/mpg123.framework/../dSYMs/mpg123.framework.dSYM/ bin/iPhone/Release/xcf_ios.app/../mpg123.framework.dSYM 
_CopyUserFrameworkDebugSymbols:
  Removing directory "bin/iPhone/Release/xcf_ios.app/../mpc.framework.dSYM".
  /usr/bin/ditto -rsrc ../../external/mpc.xcframework/ios-arm64/mpc.framework/../dSYMs/mpc.framework.dSYM/ bin/iPhone/Release/xcf_ios.app/../mpc.framework.dSYM 
_CopyUserFrameworkDebugSymbols:
  Removing directory "bin/iPhone/Release/xcf_ios.app/../MAC.framework.dSYM".
  /usr/bin/ditto -rsrc ../../external/mac.xcframework/ios-arm64/MAC.framework/../dSYMs/MAC.framework.dSYM/ bin/iPhone/Release/xcf_ios.app/../MAC.framework.dSYM 
_CopyUserFrameworkDebugSymbols:
  Removing directory "bin/iPhone/Release/xcf_ios.app/../lame.framework.dSYM".
  /usr/bin/ditto -rsrc ../../external/lame.xcframework/ios-arm64/lame.framework/../dSYMs/lame.framework.dSYM/ bin/iPhone/Release/xcf_ios.app/../lame.framework.dSYM 
_CopyUserFrameworkDebugSymbols:
  Removing directory "bin/iPhone/Release/xcf_ios.app/../flac.framework.dSYM".
  /usr/bin/ditto -rsrc ../../external/flac.xcframework/ios-arm64/flac.framework/../dSYMs/flac.framework.dSYM/ bin/iPhone/Release/xcf_ios.app/../flac.framework.dSYM 
_CopyUserFrameworkDebugSymbols:
  Removing directory "bin/iPhone/Release/xcf_ios.app/../dumb.framework.dSYM".
  /usr/bin/ditto -rsrc ../../external/dumb.xcframework/ios-arm64/dumb.framework/../dSYMs/dumb.framework.dSYM/ bin/iPhone/Release/xcf_ios.app/../dumb.framework.dSYM 
```